benches/misc/coprocessor/codec/batch/mod.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use test::Bencher;
+use tikv::coprocessor::codec::batch::*;
+use tikv::coprocessor::codec::mysql::*;
+
+#[bench]
+fn bench_column_agg_long(b: &mut Bencher) {
+    let rows = 1024;
+    let mut long_col = VectorValue::with_capacity(rows, cop_datatype::EvalType::Int);
+    for row_id in 0..rows {
+        if row_id & 1 == 0 {
+            long_col.push_int(None);
+            continue;
+        }
+        long_col.push_int(Some(row_id as i64));
+    }
+    b.iter(|| {
+        let slice = long_col.as_int_slice();
```

Comment:
I don't think it's appropriate to put `as_int_slice` *in* the iteration, because batch execution does not do it in each iteration.

See:

- `borrow_vector_value ` is placed outside the loop:  https://github.com/tikv/tikv/blob/41cdf095758585007872e5cd04fcee50354e2569/src/coprocessor/dag/rpn_expr/function.rs#L153
- `borrow_vector_value` is implemented using `VectorValue::as_ref`: https://github.com/tikv/tikv/blob/41cdf095758585007872e5cd04fcee50354e2569/src/coprocessor/codec/data_type/mod.rs#L94
- `VectorValue::as_ref` is implemented using `VectorValue::as_xxx_slice`: https://github.com/tikv/tikv/blob/41cdf095758585007872e5cd04fcee50354e2569/src/coprocessor/codec/data_type/vector.rs#L639

Reply:
And then when it is placed outside the loop, the cost of each access is really only array accessing :)